packages/adblocker/src/engine/bucket/cosmetic.ts

```diff
@@ -21,6 +21,7 @@ import { hashStrings, tokenizeNoSkip } from '../../utils';
 import { noopOptimizeCosmetic } from '../optimizer';
 import ReverseIndex from '../reverse-index';
 import FiltersContainer from './filters';
+import { FiltersEngine } from '../../../adblocker';
```

Comment:
Let's avoid this circular dependency.

Reply:
I'm not sure how to deal with this because importing this function allows us to enforce the proper types. Would you like to types in a separate file?

Reply:
Resolved by removing event emission from the cosmetic bucket.